src/Gui/ViewProviderDocumentObject.cpp

```diff
@@ -72,17 +72,19 @@ ViewProviderDocumentObject::ViewProviderDocumentObject()
   : pcObject(0)
   , pcDocument(0)
 {
-    ADD_PROPERTY(DisplayMode,((long)0));
-    ADD_PROPERTY(Visibility,(true));
-    ADD_PROPERTY(ShowInTree,(true));
-
-    ADD_PROPERTY(SelectionStyle,((long)0));
+    static const char *dogroup = "Display Options";
+    static const char *sgroup = "Selection";
+    
+    ADD_PROPERTY_TYPE(DisplayMode, ((long)0), dogroup, App::Prop_None, "Set the display mode");
+    ADD_PROPERTY_TYPE(Visibility, (true), dogroup, App::Prop_None, "Show the object in the 3d view");
+    ADD_PROPERTY_TYPE(ShowInTree, (true), dogroup, App::Prop_None, "Show the object in the tree view");
+    
+    ADD_PROPERTY_TYPE(SelectionStyle, ((long)0), sgroup, App::Prop_None, "Set the object selection style");
     static const char *SelectionStyleEnum[] = {"Shape","BoundBox",0};
     SelectionStyle.setEnums(SelectionStyleEnum);
 
     static const char* OnTopEnum[]= {"Disabled","Enabled","Object","Element",NULL};
-    ADD_PROPERTY(OnTopWhenSelected,((long int)0));
-    ADD_PROPERTY_TYPE(OnTopWhenSelected,((long int)0), "Base", App::Prop_None, 
+    ADD_PROPERTY_TYPE(OnTopWhenSelected,((long int)0), sgroup, App::Prop_None,

```

Comment:
This property assignment was already duplicated. The result was the tooltip not correctly added to the property. I just left the ADD_PROPERTY_TYPE statement.